### PR TITLE
Add Logger class and tests to VSIX.

### PR DESCRIPTION
**Pull request by bigfluffycookie**

### CODE REVIEW

File: Exercise/VSIX.UnitTests/LoggerTests.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.VisualStudio.Shell.Interop;
```
Comment (bigfluffycookie) [inline Exercise/VSIX.UnitTests/LoggerTests.cs:1]: Exact same as in VSIX just copied it over.

File: Exercise/VSIX.UnitTests/VSIX.UnitTests.csproj [hunk 2]
```diff
@@ -1,19 +1,25 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>net48</TargetFramework>
     <ImplicitUsings>disable</ImplicitUsings>
 	  <LangVersion>latest</LangVersion>
-    <Nullable>enable</Nullable>
+    <Nullable>disable</Nullable>
```
Comment (bigfluffycookie) [inline Exercise/VSIX.UnitTests/VSIX.UnitTests.csproj:7]: I forgot to disable this earlier so I did it here as the code in the unit tests (??=) complained.

File: Exercise/VSIX/ILogger.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.VisualStudio.Shell;
```
Comment (bigfluffycookie) [inline Exercise/VSIX/ILogger.cs:1]: Exactly the same as in Step5 just copied it

**Review by duncanp-sonar — APPROVED** [review]
Looks good.
Any idea why the tests passed but the build still failed?

**Comment by duncanp-sonar** [comment]
@bigfluffycookie [this action](https://github.com/marketplace/actions/setup-msbuild) might help.

**Pull request merged**